src/fs.rs

```diff
@@ -0,0 +1,513 @@
+//! Access the file system.
+//!
+//! The APIs must be added to `tauri.allowlist.fs` in `tauri.conf.json`:
+//! ```json
+//! {
+//!   "tauri": {
+//!     "allowlist": {
+//!       "fs": {
+//!         "all": true, // enable all FS APIs
+//!         "readFile": true,
+//!         "writeFile": true,
+//!         "readDir": true,
+//!         "copyFile": true,
+//!         "createDir": true,
+//!         "removeDir": true,
+//!         "removeFile": true,
+//!         "renameFile": true,
+//!         "exists": true
+//!       }
+//!     }
+//!   }
+//! }
+//! ```
+//! It is recommended to allowlist only the APIs you use for optimal bundle size and security.
+use crate::Error;
+use js_sys::ArrayBuffer;
+use serde::{Deserialize, Serialize};
+use serde_repr::*;
+use std::path::{Path, PathBuf};
+use std::str;
+
+#[derive(Serialize_repr)]
+#[repr(u16)]
+pub enum BaseDirectory {
+    Audio = 1,
+    Cache = 2,
+    Config = 3,
+    Data = 4,
+    LocalData = 5,
+    Desktop = 6,
+    Document = 7,
+    Download = 8,
+    Executable = 9,
+    Font = 10,
+    Home = 11,
+    Picture = 12,
+    Public = 13,
+    Runtime = 14,
+    Template = 15,
+    Video = 16,
+    Resource = 17,
+    App = 18,
+    Log = 19,
+    Temp = 20,
+    AppConfig = 21,
+    AppData = 22,
+    AppLocalData = 23,
+    AppCache = 24,
+    AppLog = 25,
+}
+
+#[derive(Deserialize)]
```

Comment:
Can we have these implement Debug, Eq, PartialEq, Clone

Reply:
For me it's fine. Would like to get @JonasKruckenberg input before adding it though.

Reply:
I see no reason to object here, sounds good